### ISSUE TITLE
List-Unsubscribe header not in double opt-in confirmation

### ISSUE THREAD

**Issue by alt-romes** [knadh/listmonk#2224]
**Version:**
 - listmonk: v4.1.0
 - OS: NixOS

**Description of the bug and steps to reproduce:**

It seems like the List-Unsubscribe feature is broken, or at least not properly configured by default (even when the option is toggled).

I'm using listmonk v4.1 and the emails sent by the campaign do not include the List-Unsubscribe header at all, and I have tried both with and without the option being toggled.

**Screenshots:**
I would upload a screenshot, but it would be hard to parse. The point is: with the option to include the list-unsubscribe header toggled on, and looking at the received email headers, the message does not have the list-unsubscribe header anywhere.

Any help would be very welcome.

**Comment by knadh** [comment]
@alt-romes if the toggle is enabled, then listmonk definitely includes the header. What SMTP server are you using? It may be possible that the SMTP server is stripping it?

**Comment by alt-romes** [comment]
> @alt-romes if the toggle is enabled, then listmonk definitely includes the header. What SMTP server are you using? It may be possible that the SMTP server is stripping it?

I'm using postfix. That's a good hypothesis. I'll troubleshoot that in particular, thanks for the quick deploy

**Comment by alt-romes** [comment]
I tried using a simple smtpd server to log the messages (`python3 -u -m smtpd -n -c DebuggingServer 127.0.0.1:1025`), and it confirmed the mail sent from Listmonk doesn't have a List-Unsubscribe header:

```
---------- MESSAGE FOLLOWS ----------
mail options: ['BODY=8BITMIME']
b'Mime-Version: 1.0'
b'Content-Type: text/html; charset=UTF-8'
b'Content-Transfer-Encoding: quoted-printable'
b'To: <rodrigo@kanjideck.com>'
b'Subject: Confirm subscription'
b'Message-Id: <1735314136278694511.1085.2907996738875964026@mogbit>'
b'From: "Kanjideck Newsletter" <subscribe@kanjideck.com >'
b'Date: Fri, 27 Dec 2024 15:42:16 +0000'
b'X-Peer: 127.0.0.1'
b''
b''
b''
b'<!doctype html>'
b'<html>'
b'    <head>'
....
------------ END MESSAGE ------------
```

I also grepped a bit, and I found out that while I was setting up listmonk I did send out a few test e-mails that did contain the headers for a day or two, but they stopped showing up in all mails that came after.

Is there any other test you'd like me to run? It can also be the configuration of the smtpd connection?

Thanks again

Configuration of main server:
<img width="1125" alt="Screenshot 2024-12-27 at 15 47 52" src="https://github.com/user-attachments/assets/c7e2187e-980f-43b8-b2fc-f4caa0512db3" />

Configuration of dummy debug server:
<img width="1089" alt="Screenshot 2024-12-27 at 15 48 34" src="https://github.com/user-attachments/assets/a78e5675-ca61-4ed8-a7bf-3ab0cf99dc74" />

**Comment by alt-romes** [comment]
Right, so I've figured out the actual problem, which I misdiagnosed:

The List-Unsubscribe headers are not added to the *Double-opt-in* confirmation message.

They work as expected when campaigns are sent.
